bioptim/optimization/solution.py

```diff
@@ -624,13 +626,22 @@ def __perform_integration(
 
                     else:
                         if keep_intermediate_points:
-                            integrated = np.array(nlp.dynamics[s](x0=x0, p=u, params=params / param_scaling)["xall"])
+                            if isinstance(nlp.ode_solver, OdeSolver.CVODES):
+                                raise NotImplementedError("keep_intermediate_points is not implemented with CVODES")
```

Comment:
This if-statement should be done outside the loop so it does not slow down the integration by checking again and again the same thing